src/ccnl-cs/src/ccnl-cs-ll.c

```diff
@@ -0,0 +1,282 @@
+static ccnl_cs_content_t *ll = NULL;
+
+/**
+ * The struct holds a list of \ref dispatchFct and 
+ * \ref cMatchFct function pointers per packet types. 
+ *
+ * TODO: refactor for lose coupling
+ */
+extern struct ccnl_suite_s ccnl_core_suites[CCNL_SUITE_LAST];
+
+/**
+ * Compares two \ref ccnl_cs_content_t elements
+ *
+ * @param[in] a The first item 
+ * @param[in] b The second item
+ *
+ * @return 0 if the first item \ref a is equal to the second item \ref b
+ * @return -1 if the first item \ref a should sort before the second item \ref b
+ * @return 1 if the first item \ref a should sort after the second item \ref b
+ */
+//static int _content_cmp(const ccnl_cs_content_t *a, const ccnl_cs_content_t *b);
+
+static int _content_cmp_name(const ccnl_cs_content_t *a, const ccnl_cs_content_t *b);
+
+static ccnl_cs_status_t ccnl_cs_ll_add(const ccnl_cs_name_t *name, const ccnl_cs_content_t *content);
+static ccnl_cs_status_t ccnl_cs_ll_lookup(const ccnl_cs_name_t *name, ccnl_cs_content_t *content);
+static ccnl_cs_status_t ccnl_cs_ll_remove(const ccnl_cs_name_t *name);
+static ccnl_cs_status_t ccnl_cs_ll_clear(void);
+static ccnl_cs_status_t ccnl_cs_ll_dump(void);
+static ccnl_cs_status_t ccnl_cs_ll_age(void);
+static ccnl_cs_status_t ccnl_cs_ll_remove_oldest_entry(void);
+static int ccnl_cs_ll_match_interest(const struct ccnl_pkt_s *packet, ccnl_cs_content_t *content);
+
+static bool ccnl_cs_ll_exists(ccnl_cs_name_t *name);
+
+/**
+ * Compares two content store entries.
+ *
+ * The function checks if an element \ref a in the content store matches an element \ref b. Particulary,
+ * the function first checks if the packet types match (since the CS can handle different packet types
+ * at the same time) and then passes the items to a \ref cMatchFct function. This function is unique per 
+ * packet type. See also \ref ccnl_suite_s. This function is only used for \ref ccnl_fwd_handleInterest
+ *
+ * @param[in] a The first item 
+ * @param[in] b The second item
+ *
+ * @return ?
+ */
+static int ccnl_cs_ll_cmp_match_interest(const ccnl_cs_content_t *a, const ccnl_cs_content_t *b);
+
+void ccnl_cs_init_ll(ccnl_cs_ops_t *ccnl_cs_ops_ll) {
+    /* set the function pointers */
+    ccnl_cs_init(ccnl_cs_ops_ll, ccnl_cs_ll_add, ccnl_cs_ll_lookup, ccnl_cs_ll_remove, ccnl_cs_ll_clear, 
+        ccnl_cs_ll_dump, ccnl_cs_ll_age, NULL, ccnl_cs_ll_remove_oldest_entry, ccnl_cs_ll_match_interest);
+}
+
+/*
+static int _content_cmp(const ccnl_cs_content_t *a, const ccnl_cs_content_t *b) {
+    (void)a;
+    (void)b;
+    return 0;
+}
+*/
+
+static int _content_cmp_name(const ccnl_cs_content_t *a, const ccnl_cs_content_t *b) {
+    /** do the number of components match */
+    if (a->pkt->pfx->compcnt == b->pkt->pfx->compcnt) {
+        /** does the length of the names match */
+        if (a->pkt->pfx->namelen== b->pkt->pfx->namelen) {
+            /** compare the names */
+            return memcmp(a->pkt->pfx->nameptr, b->pkt->pfx->nameptr, a->pkt->pfx->namelen);
```

Comment:
is it guaranteed that name components are sequentially stored? I am not sure whether a simple memcmp is working here..